e2e/test/scenarios/embedding/embedding-reproductions.cy.spec.js

```diff
@@ -504,106 +501,167 @@ describe("issues 20845, 25031", () => {
+      );
+    });
+  });
+
+  describe("should allow a dashboard filter to map to a boolean field filter parameter (metabase#27643)", () => {
+    beforeEach(() => {
+      const dashboardParameter = {
+        id: "2850aeab",
+        name: "Text filter for boolean field",
+        slug: "text_filter_for_boolean_field",
+        type: "string/=",
+      };
+
+      const dashboardDetails = {
+        name: "Dashboard with card with boolean field filter",
+        enable_embedding: true,
+        embedding_params: {
+          [dashboardParameter.slug]: "enabled",
         },
+        parameters: [dashboardParameter],
       };
-      createNativeQuestion(questionDetails)
-        .then(({ body: question }) => {
-          const DASHBOARD_FILTER_ID = "2850aeab";
-          const DASHBOARD_PARAMETER = createMockParameter({
-            id: DASHBOARD_FILTER_ID,
-            name: "Text filter for boolean field",
-            slug: "text_filter_for_boolean_field",
+
+      cy.get("@postgresInvoicesExpectedInvoiceId")
+        .then(fieldId => {
+          cy.createNativeQuestionAndDashboard({
+            questionDetails: getQuestionDetails(fieldId),
+            dashboardDetails,
           });
-          const dashboardDetails = {
-            name: "dashboard with card with boolean field filter",
-            enable_embedding: true,
-            embedding_params: {
-              [DASHBOARD_PARAMETER.slug]: "enabled",
-            },
-            dashcards: [
-              createMockDashboardCard({
-                id: -1,
-                size_x: 12,
-                size_y: 8,
-                card_id: question.id,
-                parameter_mappings: [
-                  {
-                    parameter_id: DASHBOARD_FILTER_ID,
-                    card_id: question.id,
-                    target: ["dimension", ["template-tag", TEMPLATE_TAG_NAME]],
-                  },
-                ],
-              }),
+        })
+        .then(({ body: dashboardCard }) => {
+          const { card_id, dashboard_id } = dashboardCard;
+
+          cy.wrap(dashboard_id).as("dashboardId");
+          cy.wrap(card_id).as("questionId");
+
+          const mapFilterToCard = {
+            parameter_mappings: [
+              {
+                parameter_id: dashboardParameter.id,
+                card_id,
+                target: ["dimension", ["template-tag", TEMPLATE_TAG_NAME]],
+              },
             ],
-            parameters: [DASHBOARD_PARAMETER],
           };
-          return createDashboardWithTabs(dashboardDetails);
-        })
-        .then(dashboard => {
-          cy.wrap(dashboard.id).as("dashboardId");
+
+          cy.editDashboardCard(dashboardCard, mapFilterToCard);
+        });
+    });
+
+    it("in static embedding and in public dashboard scenarios (metabase#27643-1)", () => {
+      cy.log("Test the dashboard");
+      visitDashboard("@dashboardId");
+      getDashboardCard().should("contain", "true");
```

Comment:
This is brilliant. At first, I thought there was no way this would work, but then I realized since we limit the result to 1. We can expect that it would be filtered correctly.

Reply:
Well, yes.
SInce we're already dealing with the binary choice, I thought let's limit it to 1 so it can be either true or false.